subgraphs/exchange/exchange.graphql

```diff
@@ -93,6 +93,106 @@ type SubsidizedSwapped @entity {
   path: [Bytes!]
 }
 
+# For LimitOrder contract
+type LimitOrderFilledByProtocol @entity(immutable: true) {
+  id: ID!
+  orderHash: Bytes! # bytes32
+  maker: Bytes! # address
+  taker: Bytes! # address
+  allowFillHash: Bytes! # bytes32
+  relayer: Bytes! # address
+  profitRecipient: Bytes! # address
+  fillReceipt_makerToken: Bytes! # address
+  fillReceipt_takerToken: Bytes! # address
+  fillReceipt_makerTokenFilledAmount: BigInt! # uint256
+  fillReceipt_takerTokenFilledAmount: BigInt! # uint256
+  fillReceipt_remainingAmount: BigInt! # uint256
+  fillReceipt_makerTokenFee: BigInt! # uint256
+  fillReceipt_takerTokenFee: BigInt! # uint256
+  takerTokenProfit: BigInt! # uint256
+  takerTokenProfitFee: BigInt! # uint256
+  takerTokenProfitBackToMaker: BigInt! # uint256
+  blockNumber: BigInt!
+  blockTimestamp: BigInt!
+  transactionHash: Bytes!
+}
+
+type LimitOrderFilledByTrader @entity(immutable: true) {
+  id: ID!
+  orderHash: Bytes! # bytes32
+  maker: Bytes! # address
+  taker: Bytes! # address
+  allowFillHash: Bytes! # bytes32
+  recipient: Bytes! # address
+  fillReceipt_makerToken: Bytes! # address
+  fillReceipt_takerToken: Bytes! # address
+  fillReceipt_makerTokenFilledAmount: BigInt! # uint256
+  fillReceipt_takerTokenFilledAmount: BigInt! # uint256
+  fillReceipt_remainingAmount: BigInt! # uint256
+  fillReceipt_makerTokenFee: BigInt! # uint256
+  fillReceipt_takerTokenFee: BigInt! # uint256
+  blockNumber: BigInt!
+  blockTimestamp: BigInt!
+  transactionHash: Bytes!
+}
+
+type OrderCancelled @entity(immutable: true) {
+  id: ID!
+  orderHash: Bytes! # bytes32
+  maker: Bytes! # address
+  blockNumber: BigInt!
+  blockTimestamp: BigInt!
+  transactionHash: Bytes!
+}
+
+enum OrderStatus {
+  Normal
+  Cancelled
+  FullyFilled
+}
+
+enum LimitOrderTypes {
+  ByProtocol
+  ByTrader
+}
+
+type Order @entity(immutable: false) {
+  id: ID! # orderHash
+  limitOrderFilledId: [String!]! # One-to-many relationship
+  orderStatus: OrderStatus!
+  maker: Bytes! # address
+  makerToken: Bytes! # address
+  takerToken: Bytes! # address
+  firstFilledTime: BigInt!
+  lastFilledOrCancelledTime: BigInt!
```

Comment:
Is this possible to be separated into two field, `lastFilledTime` and `cancelledTime`? I think it's more clear to record them respectively.

Reply:
OK